### PR TITLE
chore(conport): quarantine dead codebases

### PR DESCRIPTION
**Pull request by hu3mann**
## Summary

Completes DMX-CONPORT-OPTIMAL packet 103 by marking non-canonical ConPort code surfaces as quarantined without deleting code.

## Changes

- Adds a deprecation notice to `src/conport/memory_server.py`
- Adds `services/conport_kg/QUARANTINE.md`
- Adds a canonical-codebase note to `docker/mcp-servers-source/conport/SURFACE_INVENTORY.md`
- Adds packet proof at `proof/conport-optimal-103/PROOF.md`

## Validation

- Packet JSON parse/schema validation passed
- `python3 -m py_compile src/conport/memory_server.py` passed
- Runtime import grep for `src.conport.memory_server` found no runtime imports
- Runtime import grep for `services.conport_kg` found one test-only import, recorded in proof
- `git diff --check` passed
- file-scoped pre-commit passed
- Task Orchestrator packet 103 moved to terminal

## Notes

No files were deleted. No runtime code in the canonical ConPort server was changed.

### CODE REVIEW

**Comment by hu3mann** [comment]
## 🔍 Deep review — automated (Claude Opus)

**Verdict: REQUEST CHANGES** — the `memory_server.py` quarantine is sound, but the **`services/conport_kg/` "dead" claim is not supported by the evidence; there are live, compose-wired runtime consumers.** (PR is DRAFT — flagging before it leaves draft.)

This is a docs-only change (94 add / 0 del, no code/packaging modified), so build/test/import-break risk for *this diff* is near zero. The risk is **truth of the central claim** and **downstream contradiction**, which the repo's own governance treats as the thing to verify.

---

### 🔴 BLOCKING — `services/conport_kg/` is not dead; it has live runtime callers (confidence: high)

The PR proof asserts the only `conport_kg` reference is a single test-only import and that the module "is not present" in this checkout. That is incomplete. The grep used (`from conport_kg` / `import conport_kg`) **misses `sys.path.insert`-style runtime imports**, which is how every real consumer loads it:

```
services/serena/kg_provider.py:21              sys.path.insert(0, .../conport_kg) → from queries.overview import OverviewQueries ...
services/dopecon-bridge/kg_endpoints.py:29     sys.path.insert(0, .../conport_kg) → from queries.* import ...
services/mcp-integration-bridge/kg_endpoints.py:21   sys.path.insert(0, .../conport_kg) → from queries.* import ...
services/agents/cognitive_guardian_kg.py:40    sys.path.insert(0, .../conport_kg)
services/dddpg/kg_integration.py:34            sys.path.insert(0, .../conport_kg)
```

These import `queries.overview/exploration/deep_context/models`, which **do exist** (`services/conport_kg/queries/{overview,exploration,deep_context,models}.py`). And they are wired into running services:

- `services/mcp-integration-bridge/main.py:1239` actively does `from kg_endpoints import router as kg_router` → `app.include_router(kg_router)` (registers `/kg/*`).
- `dopecon-bridge` and `serena` are **compose-wired services** (`compose.yml:389` `dopecon-bridge:`, `compose.yml:541` `serena:`).
- `services/conport_kg` even has its own deployment: `docker/conport-kg/docker-compose.yml` (per `docker/conport-kg/README.md`).

The imports are wrapped in `try/except ImportError`, so they degrade gracefully rather than hard-crash — but that means `conport_kg` is a **soft runtime dependency**, not dead code. Stamping it `QUARANTINE.md` ("Do not import, deploy, or route runtime traffic") directly contradicts code that imports and routes to it today.

This is also already documented in the repo: `proof/TP-DMX-EVIDENCE-GATE-VERIFY-001/VERIFICATION_GATES_REPORT.json:39` lists exactly these `sys.path.insert` callers and classifies `conport_kg` as **"(C) Utility/library"**, not "(E) Empty/dead". The PR's deadness conclusion conflicts with prior in-repo evidence and doesn't address it.

**Required before merge:** either (a) walk each caller and show it's itself dead/unreachable (not done in the proof), or (b) reclassify — `conport_kg` is a quarantine *candidate* with live consumers that must be cut over first. Do not assert "dead" on the current evidence.

---

### 🟡 IMPORTANT — contract-sensitive surface not reconciled: `config/runtime_authority_manifest.json` (confidence: high)

`memory_server.py` deadness itself checks out (no compose/Dockerfile/entrypoint/static import — see below). But the runtime authority manifest still encodes it as a *live known-conflict*:

- `config/runtime_authority_manifest.json:255` lists `src/conport/memory_server.py` under `forbidden_authority_paths` and `:265-277` under `known_conflicts` ("deployment authority is not collapsed").

If the project's position is now "this is dead, canonical is `docker/mcp-servers-source/conport/`", the manifest (a contract-sensitive surface per governance) should be updated in the same change, or the PR should explicitly state why it's deferred. Right now the new `SURFACE_INVENTORY.md` "Canonical Codebase" section and the manifest tell two different stories.

---

### 🟢 MINOR — `scripts/brand_lint.py` references the quarantined file (confidence: high, low impact)

`scripts/brand_lint.py:35` includes `"src/conport/memory_server.py"` in `STRICT_LOG_FILES`; brand_lint reads/AST-parses it. The added deprecation docstring doesn't break the AST parse (verified conceptually; it's a docstring prepend), so no lint break expected — but be aware the file is still actively linted, i.e. it is not as "retired" from the toolchain as the notice implies. No change strictly required.

---

### 🟢 MINOR — proof self-flags the same risk (confidence: high)

The proof's own "Residual Risk" admits S5 edits a file under the directory an invariant says is canonical/untouchable, rationalized as a "docs-only exception." That's a reasonable call for `SURFACE_INVENTORY.md`, but it does mean the packet's invariants and steps are in tension — worth an operator nod given ConPort is contract-sensitive.

---

### What checks out (PASS)

- **`src/conport/memory_server.py` is genuinely unreferenced as runtime:** no `docker/` or `compose.yml` reference, no static `import`/`from` of `conport.memory_server` anywhere live. Quarantine-by-docstring for *this file* is defensible. (PASS)
- **No deletions, no runtime code changed** — diff is docstring + 2 new markdown files + proof. Reversible (just delete the notes / revert docstring). (PASS)
- **Packet/schema validation, py_compile, `git diff --check`** as claimed — not re-run here (NOT_RUN), but plausible for a docs diff.

---

### Validation summary
- `memory_server.py` runtime-dead claim: **PASS** (independently verified — no compose/docker/static-import references)
- `conport_kg` runtime-dead claim: **FAIL** (5 live `sys.path.insert` callers; 2+ compose-wired; `mcp-integration-bridge` registers its router; own `docker/conport-kg` deploy; contradicts `VERIFICATION_GATES_REPORT.json`)
- runtime_authority_manifest reconciliation: **NOT_RUN by PR** (stale relative to the new canonical claim)
- py_compile / schema / pre-commit: **NOT_RUN** (trusting PR claims for docs-only diff)

**Bottom line:** Split the claim. The `memory_server.py` quarantine can stand. The `conport_kg` quarantine is premature and currently *false* as written — it needs either a caller-deadness walk or a downgrade from "dead" to "quarantine candidate, consumers pending cutover," plus manifest reconciliation. Posting on a DRAFT so it can be fixed before review.

File: services/conport_kg/QUARANTINE.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Quarantine Notice
+
+`services/conport_kg/` is a non-canonical Apache AGE graph projection.
+
+Per the Memory Trinity ADR, this directory is not the runtime-integrated
+ConPort MCP server. The canonical ConPort MCP server lives at:
+
+`docker/mcp-servers-source/conport/`
+
+Do not import, deploy, or route runtime traffic to this code without explicit
```
Comment (hu3mann) [inline services/conport_kg/QUARANTINE.md:10]: Accurate as written for *deployment* (none of the `conport_kg` consumers are compose-wired: `mcp-integration-bridge` and `dddpg`/`agents` are not in `compose.yml`, and the compose-wired `dopecon-bridge`/`serena` do not route to `conport_kg`). But note the directory is **not** free of code-level callers: `services/{serena/kg_provider.py:21, mcp-integration-bridge/kg_endpoints.py:21, dopecon-bridge/kg_endpoints.py:29, agents/cognitive_guardian_kg.py:40, dddpg/kg_integration.py:34}` all `sys.path.insert` into this tree and import `queries.*` (which exist here). The quarantine conclusion holds, but the proof should enumerate these and show each is non-deployed rather than asserting 'NONE'.

**Review by hu3mann — COMMENTED** [review]
> 🤖 **Automated thorough review** — Claude Code (Opus 4.8). Verdict: **comment** · Blocking issues: **0**. Inline comments anchor the key findings; full analysis + cross-PR conflicts below.

## Overview
PR #957 completes DMX-CONPORT-OPTIMAL-103 by quarantining (not deleting) two non-canonical ConPort surfaces: it prepends a `DEAD CODE` notice to `src/conport/memory_server.py`, adds `services/conport_kg/QUARANTINE.md`, adds a `Canonical Codebase` section to `docker/mcp-servers-source/conport/SURFACE_INVENTORY.md`, and adds `proof/conport-optimal-103/PROOF.md`. No runtime code or packaging changed; the change is fully reversible. CI is green.

This review independently verified the deadness claims against the working tree and against prior in-repo evidence, which is genuinely conflicting. The PR's *conclusion* is sound and conservative; its *proof* is weak.

## Findings

### 🟡 IMPORTANT — proof's `conport_kg` deadness evidence is incomplete (`proof/conport-optimal-103/PROOF.md:55-68`)
The proof's grep (`from conport_kg` / `import conport_kg`) misses every real consumer, all of which load `conport_kg` via `sys.path.insert`, not direct package import:
- `services/serena/kg_provider.py:21`, `services/mcp-integration-bridge/kg_endpoints.py:21`, `services/dopecon-bridge/kg_endpoints.py:29`, `services/agents/cognitive_guardian_kg.py:40`, `services/dddpg/kg_integration.py:34` — all do `sys.path.insert(..., conport_kg)` and import `queries.{overview,exploration,deep_context,models}`, which **do exist** under `services/conport_kg/queries/`.
The proof concludes "NONE / test-only" without ever surfacing these. It reaches a defensible answer for the wrong reasons.

**However, the conclusion still holds** after deployment analysis (verified):
- `mcp-integration-bridge` (the only module that registers a kg router) is **NOT in `compose.yml`** — not deployed.
- `dopecon-bridge` (compose-wired) does **NOT** route to `conport_kg`: its runtime `dopecon_bridge/routes.py:129` defines its own `kg_router` proxying to the ConPort client; `main.py:72` includes that, not `kg_endpoints.py`.
- `serena` (compose-wired) has **zero** runtime references to `kg_provider`; `dddpg`/`agents`/`dope-query` are not compose-wired.
This matches the repo's own `proof/TP-DMX-EVIDENCE-GATE-VERIFY-001/VERIFICATION_GATES_REPORT.json`, which classified `conport_kg` as "inactive/superseded". **Ask:** strengthen the proof to (a) list the `sys.path.insert` callers, (b) show each is non-deployed/non-wired, and (c) cite VG-011. Deadness should rest on deployment evidence, not on a grep that didn't even find the callers.

### 🟡 IMPORTANT — contract-sensitive manifest not reconciled: `config/runtime_authority_manifest.json`
`memory_server.py` deadness checks out (no compose/docker/static-import references). But `runtime_authority_manifest.json:255` lists `src/conport/memory_server.py` under `forbidden_authority_paths` and `:274-277` as a live `known_conflict`. The new `SURFACE_INVENTORY.md` "Canonical Codebase" claim and this manifest now tell two different stories. ConPort surfaces are contract-sensitive; either reconcile the manifest in this PR or state explicitly why it is deferred.

### 🟢 MINOR — quarantined file still actively linted (`scripts/brand_lint.py:35`)
`src/conport/memory_server.py` remains in `STRICT_LOG_FILES`; brand_lint AST-parses it. The prepended docstring doesn't break parsing, so no break — but the file is not as "retired" from the toolchain as the notice implies. No change required.

### 🟢 MINOR — proof self-flags S5 invariant tension (`PROOF.md:63-69`)
The proof's own Residual Risk admits S5 edits `SURFACE_INVENTORY.md` under a directory an invariant calls canonical/untouchable, rationalized as a docs-only exception. Reasonable, but worth an operator nod given contract-sensitivity.

## Test coverage
No tests in scope (docs-only). Quarantine mechanism is documentation, not code, so no test coverage is expected or added. Reversibility is structurally guaranteed (revert the docstring + delete two notices).

## Validation
- `memory_server.py` runtime-dead claim: **PASS** (independently verified — no compose/docker/static-import references)
- `conport_kg` runtime-dead-as-deployed claim: **PASS** (no consumer is compose-wired or runtime-wired), but proof EVIDENCE is **incomplete** (FAIL on rigor — grep missed all callers)
- `runtime_authority_manifest.json` reconciliation: **NOT_RUN by PR** (stale relative to the new canonical claim)
- CI: **PASS** (all required checks green; 3 environment-gated checks skipped)

## Verdict
**comment.** The change is reversible, docs-only, CI-green, and its conclusion is defensible and more conservative than the repo's own prune recommendation. Not blocking. But before merge the proof should be hardened to surface the `sys.path.insert` consumers and ground deadness in deployment evidence, and the `runtime_authority_manifest.json` conflict should be reconciled or explicitly deferred. (A prior automated review rated `conport_kg` deadness a BLOCKING FAIL by counting caller *files*; that overweights code presence and underweights deployment reality — none of those callers are deployed/wired, so the quarantine itself is correct.)

---

### 🔗 Cross-PR analysis

**#958** (conport-optimal-104, "expose custom data MCP tools") touches a DISJOINT file set — `docker/mcp-servers-source/conport/{server.py, enhanced_server.py, conport_mcp_stdio.py}` (the canonical runtime surface) plus new tests/proofs. **No file overlap** with #957 (which only touches `src/conport/memory_server.py`, `services/conport_kg/QUARANTINE.md`, `SURFACE_INVENTORY.md`, and a proof dir). No merge conflict. There is a logical ordering *preference*: #957 establishes `docker/mcp-servers-source/conport/` as the single canonical ConPort codebase, and #958 extends exactly that codebase — so #957 first reinforces #958's premise, but they are independently mergeable in either order. Other listed PRs (933,943,949,966,967,968,970,971) show no file overlap with #957's 4 files.
